include/sparrow/fixed_size_layout.hpp

```diff
@@ -289,20 +290,20 @@ namespace sparrow
     auto fixed_size_layout<T>::has_value(size_type i) -> bitmap_reference
     {
         assert(i < size());
-        return m_data.bitmap[i + m_data.offset];
+        return p_data->bitmap[i + p_data->offset];
     }
 
     template <class T>
     auto fixed_size_layout<T>::has_value(size_type i) const -> bitmap_const_reference
     {
         assert(i < size());
-        return m_data.bitmap[i + m_data.offset];
+        return static_cast<const array_data*>(p_data)->bitmap[i + p_data->offset];
```

Comment:
Is this a `const_cast`? not sure why there is a need for a cast here

Reply:
I don't think so, usually `const_cast` is used to convert a const type to a non const one.
This is required otherwise `p_data` being `array_data*`, `p_array_data->bitmap[...]` return a `bitmap_reference` instead of a `bitmap_const_reference`.

Reply:
> I don't think so, usually const_cast is used to convert a const type to a non const one.

Yes but it's also sending the right message about what the cast is about, whatever the direction.

I see it's the pointed type that you need to make const ok, then I would prefer something clearer like:
```c++
const auto& readonly_data = *p_data;
return readonly_data.bitmap[i+readonly_data.offset];
```
That way the intention is clear. `static_cast` here sends the wrong message.

Reply:
Agreed. Notice that with the `std::reference_wrapper` instead of the pointer, we don't have this issue anymore.